### DISCUSSION TITLE
Build Error while installing ONNX in Linux

### DISCUSSION THREAD

**Discussion by rahul3161** [microsoft/onnxruntime#16737] [Other Q&A] [answered]
In order to install OnnX library for linux after cloning the repository. git clone --recursive https://github.com/Microsoft/onnxruntime

Then ran the build.sh with the following configuration --skip-tests --parallel and --config as Release and also as MinSizeRel

But getting this error:- error: ‘const class google::protobuf::FieldOptions’ has no member named ‘unverified_lazy’

Tried searching the problem but didn't got any particular resolution to this yet availabe. Is there some tag I am missing while executing the build.sh ?

Note:- The protobuf is already updated.

Reference link:- https://github.com/onnx/onnx/discussions/5416 (Initially posted in Onxx channel)

**Comment by xadupre** [comment, marked as answer]
If protobuf is not installed, onnxruntime does compile it. It avoids any mismatch between the installed version of the version onnxruntime requires. Otherwise, you should follow the instructions detailed here: https://onnxruntime.ai/docs/build/inferencing.html#build-instructions.

**Reply by rahul3161** [reply]
Thanks for the reply, I have tried end to end with the link shared, but the problem iterates every time.
More specifically at this stage:-
[ 22%] Linking CXX static library libprotobuf.a
[ 22%] Built target libprotobuf

**Reply by Renekton** [reply]
The path of protoc tool should be absolute path.

**Reply by rahul3161** [reply]
Tried multiple things:- 
Previous error is gone now but following popped in:- 
i) Linking CXX static library 
[ 58%] Linking CXX static library libonnxruntime_optimizer.a
[ 58%] Built target onnxruntime_optimizer

Then I tried to built from:- 
But got this error, FileNotFoundError: Unable to find requirements.txt

I am quiet new to this, probably I am doing something wrong any help will be appreciated.

Not installing protobuf on my own.

**Reply by rahul3161** [reply]
The build problem is resolved thanks for the cooperation
